l2geth/core/tx_pool.go

```diff
@@ -565,8 +565,11 @@ func (pool *TxPool) validateTx(tx *types.Transaction, local bool) error {
 	}
 	// Transactor should have enough funds to cover the costs
 	// cost == V + GP * GL
-	if pool.currentState.GetBalance(from).Cmp(tx.Cost()) < 0 {
-		return ErrInsufficientFunds
+	if !rcfg.UsingOVM {
```

Comment:
Why is this skipped?

Reply:
This check is being done in the sync service. We could do the check twice, its not the largest deal but an extra db lookup

Reply:
By skipping it, we effectively reduce the intrinsic gas by 20
https://github.com/ethereum/go-ethereum/blob/97ce6dfa6d425ba4b76278627a91c7285d6d00b2/params/protocol_params.go#L94
https://github.com/ethereum/go-ethereum/blob/94451c2788295901c302c9bf5fa2f7b021c924e2/core/vm/jump_table.go#L370

Happy to add it back but we already to the check in the syncservice

Reply:
I think the optimization isn't really worth it in this case, better to avoid diffs where possible.